src/test/testing_assert.cpp

```diff
@@ -0,0 +1,191 @@
+#include "testing_assert.hpp"
```

Comment:
What is the advantage of having this in a separate file?

Reply:
got ma reasons